src/gc/impl/conservative/gc.d

```diff
@@ -255,6 +255,10 @@ class ConservativeGC : GC
     __gshared size_t line;
     __gshared char*  file;
 
+    /// Hook used for debugging application-side
+    __gshared CollectionStartHook on_collect_start;
+    __gshared CollectionEndHook on_collect_end;
```

Comment:
One point raised by @mihails-strasuns-sociomantic is that having `delegate` + `__gshared` is dangerous. Since we have the lock it should be okay, but I'm interested to know if there's something addressing this issue anywhere else.

Reply:
Well, from pure theoretical PoV it should require `shared delegate` but I still have no idea if that is even implemented to mean something in D frontend.